automation/release/version_bump.py

```
@@ -344,9 +346,51 @@ def bump(version, docker_suffix="1"):
+
+
+def pin_source_versions(flex_sha, jans_ver):
+    """Release mode: pin FLEX_SOURCE_VERSION to the released flex commit and JANS_SOURCE_VERSION
+    to the commit the jans release pinned, so each image builds the exact released sources."""
+    if not re.fullmatch(r"[0-9a-f]{40}", flex_sha or ""):
+        raise SystemExit(f"--flex-source-sha must be a 40-char commit sha, got: {flex_sha!r}")
+    jans_sha = fetch_jans_source_version(jans_ver)
+    changed = []
+    for rel in grep_files("FLEX_SOURCE_VERSION="):
+        if sub(rel, r"(FLEX_SOURCE_VERSION=)[0-9a-f]{40}", rf"\g<1>{flex_sha}"):
+            changed.append(rel)
```

Ruff (0.15.18)

[warning] Use a list comprehension to create a transformed list

(PERF401)